src/smoosh/src/smoosh_channel.erl

```diff
@@ -194,21 +235,164 @@ handle_info(check_window, State0) ->
         end,
     erlang:send_after(60 * 1000, self(), check_window),
     {noreply, FinalState};
+handle_info(start_recovery, #state{name = Name, waiting = Waiting0} = State0) ->
+    RecActive = recover(active_file_name(Name)),
+    Waiting1 = lists:foldl(
+        fun(DbName, Acc) ->
+            case couch_db:is_compacting(DbName) of
+                true ->
+                    Priority = smoosh_server:get_priority(Name, DbName),
+                    smoosh_priority_queue:in(DbName, Priority, Priority, Acc);
+                false ->
+                    Acc
+            end
+        end,
+        Waiting0,
+        RecActive
+    ),
+    _ = State0#state{waiting = Waiting1},
```

Comment:
This is noop

Reply:
Another bug also exists here. It should be `State1 = maybe_start_compaction(State0#state{waiting = Waiting1})` and then return `State1` [here](https://github.com/apache/couchdb/pull/3766/files#diff-113cd247cfe56b1aa882a37daa2856fd33bab2e2a6183034eb3a30088960dc13R260). And also add `maybe_start_compaction/1` [here](https://github.com/apache/couchdb/pull/3766/files#diff-113cd247cfe56b1aa882a37daa2856fd33bab2e2a6183034eb3a30088960dc13R279).